src/test/java/core/basesyntax/CalculatorTest.java

```diff
@@ -0,0 +1,126 @@
+package core.basesyntax;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class CalculatorTest {
+    private static final String MESSAGE_NPE = "NPE was expected!";
+    private static final String MESSAGE_ARITHMETIC_EXCTPITON = "Expected ArithmeticException";
+
+    @Test
+    public void correctSum() {
+        Assert.assertEquals(3, Calculator.calculate( -2, '+', 5));
+    }
+
+    @Test
+    public void correctSubtraction() {
+        Assert.assertEquals(-1, Calculator.calculate( 5, '-', 6));
+    }
+
+    @Test
+    public void correctMultiply() {
+        Assert.assertEquals(-6, Calculator.calculate( -3, '*', 2));
+    }
+
+    @Test
+    public void correctDivide() {
+        Assert.assertEquals(-3, Calculator.calculate( 9, '/', -3));
+    }
+
+    @Test
+    public void correctPow() {
+        Assert.assertEquals(8, Calculator.calculate( 2, '^', 3));
+    }
+
+    @Test
+    public void operatorNull() {
+        try {
+            Assert.assertNotEquals("72", Calculator.calculate( 7, null, 2));
+        } catch (NullPointerException e) {
+            return;
+        }
+        Assert.fail(MESSAGE_NPE);
+    }
+
+    @Test
+    public void divideByZero() {
+        try {
+            Assert.assertNotEquals(0, Calculator.calculate( 7, null, 2));
+        } catch (NullPointerException e) {
+            return;
+        }
+        Assert.fail(MESSAGE_NPE);
+    }
+    @Test
+    public void numberInDegreeZero() {
+        Assert.assertEquals(1, Calculator.calculate( 2, '^', 0));
```

Comment:
```suggestion
        int actual = 1;
        int expected = Calculator.calculate( 2, '^', 0);
        Assert.assertEquals(actual, expected);
```

Думаю так було б трошки наглядніше